nvflare/ha/ha_admin_cmds.py

```diff
@@ -97,50 +39,53 @@ def get_spec(self):
                     name="list_sp",
                     description="list service providers information from previous heartbeat",
                     usage="list_sp",
-                    handler_func=list_sp,
+                    handler_func=self.list_sp,
                 ),
                 CommandSpec(
                     name="get_active_sp",
                     description="get active service provider",
                     usage="get_active_sp",
-                    handler_func=get_active_sp,
+                    handler_func=self.get_active_sp,
                 ),
                 CommandSpec(
                     name="promote_sp",
                     description="promote active service provider to specified",
                     usage="promote_sp sp_end_point",
-                    handler_func=promote_sp,
+                    handler_func=self.promote_sp,
                 ),
                 CommandSpec(
                     name="shutdown_system",
                     description="shut down entire system by setting the system state to shutdown through the overseer",
                     usage="shutdown_system",
-                    handler_func=shutdown_system,
+                    handler_func=self.shutdown_system,
                 ),
             ],
         )
 
-    def list_sp(self, args, api):
-        """List service provider information based on the last heartbeat from the overseer.
+    def list_sp(self, args, ctx: CommandContext):
```

Comment:
Just a note, can be done in another PR later.
These methods don't need to be instance method as they are not using "self"